xcube/core/imgeom.py

```diff
@@ -277,6 +277,11 @@ def _compute_output_geom(dataset: xr.Dataset,
     dst_height = 1 + math.floor((src_y_max - src_y_min) / src_xy_res)
     dst_width = i_denom * ((dst_width + i_denom - 1) // i_denom)
     dst_height = j_denom * ((dst_height + j_denom - 1) // j_denom)
+
+    # Reduce height if it takes the maximum latitude over 90° (see Issue #303).
+    if src_y_min + dst_height * src_xy_res > 90.0:
+        dst_height -= 1
+
```

Comment:
What about `dst_width`?

Reply:
Copy-pasting form [my comment](https://github.com/dcs4cop/xcube/issues/303#issuecomment-638114126) on the issue:

> In the case of the `new.new_cube()` output, it happens that the width does not exceed 360°. In this case `359 / xy_res` =~ 507.70, so the rounding behaviour doesn't add enough to take `width * xy_res` over 360°. Even tweaking the oversampling value passed to `ImageGeom.from_dataset` doesn't seem to put the width over 360°; the longitudinal check has slightly more leeway since it's looking directly at the total width, whereas the latitudinal check is adding the height to `y_min`, which in this case is already offset by 0.5° from the minimum possible value of -90°. So it's probably not worth adding an explicit width correction, unless it later becomes clear that the width can indeed somehow exceed 360°.

Reply:
Ok!